### PR TITLE
Kotlin conversion: ThrowableUtil

### PR DESCRIPTION
**Pull request by sharvaniharan**

### CODE REVIEW

File: app/src/main/java/org/wikipedia/util/ThrowableUtil.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            AppError(inner.localizedMessage!!, "")
+        } else if (throwableContainsException(e, JSONException::class.java)) {
+            AppError(context.getString(R.string.error_response_malformed),
+                    inner.localizedMessage)
+        } else {
+            // everything else has fallen through, so just treat it as an "unknown" error
+            AppError(context.getString(R.string.error_unknown),
+                    inner.localizedMessage)
+        }
+        return result
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/util/ThrowableUtil.kt:70]: Would it be possible to replace this code to inline-style? e.g.:
```
return = if (isNetworkError(e)) ...
```

File: app/src/main/java/org/wikipedia/util/ThrowableUtil.kt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    @JvmStatic
+    fun isOffline(caught: Throwable?): Boolean {
+        return (caught is UnknownHostException ||
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/util/ThrowableUtil.kt:75]: Looks like the `( )` is not necessary.

File: app/src/main/java/org/wikipedia/util/ThrowableUtil.kt [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    @JvmStatic
+    fun is404(caught: Throwable): Boolean {
+        return caught is HttpStatusException && caught.code() == 404
+    }
+
+    @JvmStatic
+    fun isNetworkError(e: Throwable): Boolean {
+        return (throwableContainsException(e, UnknownHostException::class.java) ||
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/util/ThrowableUtil.kt:91]: Looks like the `( )` is not necessary.

File: app/src/main/java/org/wikipedia/util/ThrowableUtil.kt [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                throwableContainsException(e, TimeoutException::class.java) ||
+                throwableContainsException(e, SSLException::class.java))
+    }
+
+    @Deprecated("")
+    class AppError(val error: String, val detail: String?)
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/util/ThrowableUtil.kt:97]: Considering replace/rewrite this deprecated class?

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/util/ThrowableUtil.kt (hunk 4 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/util/ThrowableUtil.kt:94]: @dbrant @sharvaniharan Do you think it is still reasonable to keep it as `deprecated`? or it is fine to display the raw server messages in certain cases and remove this `deprecated` tag?

Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/util/ThrowableUtil.kt:94]: I agree we can just remove the deprecation..re-writing can be a fresh task and we dont need to deprecate till then.

Comment (dbrant) [reply app/src/main/java/org/wikipedia/util/ThrowableUtil.kt:94]: yep, let's remove the `deprecated` tag for now.

**Pull request merged**